scope3_methodology/ad_tech_platform/model.py

```diff
@@ -0,0 +1,382 @@
+            servers_processing_cookie_syncs_percentage
+            if servers_processing_cookie_syncs_percentage
+            else defaults.servers_processing_cookie_syncs_percentage
+        )
+        self.cookie_sync_distribution_ratio = (
+            cookie_sync_distribution_ratio
+            if cookie_sync_distribution_ratio
+            else defaults.cookie_sync_distribution_ratio
+        )
+
+        # Values without fallbacks
+        self.data_transfer_emissions_mt_co2e_per_month = data_transfer_emissions_mt_co2e_per_month
+        self.corporate_allocation_percentage = corporate_allocation_percentage
+        self.server_allocation_percentage = server_allocation_percentage
+        self.distribution_partners = distribution_partners
+        self.ad_tech_platform_bid_requests_processed_billion_per_month = (
+            ad_tech_platform_bid_requests_processed_billion_per_month
+        )
+        self.cookie_syncs_processed_billion_per_month = cookie_syncs_processed_billion_per_month
+
+    def get_bid_requests_processed_per_month(self) -> float:
+        return self.bid_requests_processed_billion_per_month * 1000000000
+
+    def get_server_emissions_g_co2e_per_month(self) -> float:
+        return self.server_emissions_mt_co2e_per_month * 1000000
+
+    def get_servers_processing_bid_requests_pct(self) -> float:
+        return self.servers_processing_bid_requests_percentage / 100.0
+
+    def get_servers_processing_cookie_syncs_pct(self) -> float:
+        return self.servers_processing_cookie_syncs_percentage / 100.0
+
+    def get_cookie_syncs_processed_per_month(self) -> float | None:
+        if self.cookie_syncs_processed_billion_per_month:
+            return self.cookie_syncs_processed_billion_per_month * 1000000000
+        return None
+
+    def get_corporate_allocation_decimal_fraction(self) -> float:
+        return self.corporate_allocation_percentage / 100.0
+
+    def get_servers_processing_cookie_syncs_decimal_fraction(self) -> float:
+        return self.servers_processing_cookie_syncs_percentage / 100.0
+
+    def get_servers_processing_bid_requests_decimal_fraction(self) -> float:
+        return self.servers_processing_bid_requests_percentage / 100.0
+
+    def get_bid_request_rejection_decimal_fraction(self) -> float:
+        return self.bid_request_rejection_percentage / 100.0
+
+    def get_decimal_fraction_of_bid_requests_processed_from_ad_tech_platforms(self) -> float:
+        return self.percentage_of_bid_requests_processed_from_ad_tech_platforms / 100.0
+
+    def get_data_transfer_emissions_g_co2e_per_month(self) -> float | None:
+        if self.data_transfer_emissions_mt_co2e_per_month:
+            return self.data_transfer_emissions_mt_co2e_per_month * 1000000
+        return None
+
+    def get_server_allocation_decimal_fraction(self) -> float:
+        return self.server_allocation_percentage / 100.0
+
+    def comp_external_request_ratio(self, depth: int) -> float:
```

Comment:
```suggestion
    # We assume that server-to-server requests (generally from ad tech platforms) incur core networking costs that are not accounted for in client device data transfer emissions. Therefore we need to figure out what portion of inbound requests are S2S.
    def comp_external_request_ratio(self, depth: int) -> float:
```

We probably should add a comment here with the assumption we're making and also rename something that is more like server-to-server requests. Unclear if we should be returning the ratio or just the number of S2S requests?